### ISSUE TITLE
Return index for each result of NamedAPIResourceList

### ISSUE THREAD

**Issue by wpmonty** [bagelbits/5e-srd-api#16]
For any List, the results array contains objects that do not have an index.

Ex: http://www.dnd5eapi.co/api/ability-scores/ would return
``` json
{
    "count": 6,
    "results": [
        {
            "name": "STR",
            "url": "http://www.dnd5eapi.co/api/ability-scores/1"
        },
        {
            "name": "DEX",
            "url": "http://www.dnd5eapi.co/api/ability-scores/2"
        },
        {
            "name": "CON",
            "url": "http://www.dnd5eapi.co/api/ability-scores/3"
        },
        {
            "name": "INT",
            "url": "http://www.dnd5eapi.co/api/ability-scores/4"
        },
        {
            "name": "WIS",
            "url": "http://www.dnd5eapi.co/api/ability-scores/5"
        },
        {
            "name": "CHA",
            "url": "http://www.dnd5eapi.co/api/ability-scores/6"
        }
    ]
}
```

where instead it should return

``` json
{
    "count": 6,
    "results": [
        {
            "index": 1,
            "name": "STR",
            "url": "http://www.dnd5eapi.co/api/ability-scores/1"
        },
        {
            "index": 2,
            "name": "DEX",
            "url": "http://www.dnd5eapi.co/api/ability-scores/2"
        },
        {
            "index": 3,
            "name": "CON",
            "url": "http://www.dnd5eapi.co/api/ability-scores/3"
        },
        {
            "index": 4,
            "name": "INT",
            "url": "http://www.dnd5eapi.co/api/ability-scores/4"
        },
        {
            "index": 5,
            "name": "WIS",
            "url": "http://www.dnd5eapi.co/api/ability-scores/5"
        },
        {
            "index": 6,
            "name": "CHA",
            "url": "http://www.dnd5eapi.co/api/ability-scores/6"
        }
    ]
}
```
This will improve looped queries; otherwise it needs to parsed out from the "url" key.

**Comment by bagelbits** [comment]
This might want to wait until this PR lands:
https://github.com/bagelbits/5e-database/pull/130

**Comment by bagelbits** [comment]
I rolled it into [this PR](#26) which will get rolled out at the same time of the  above PR.

**Comment by wpmonty** [comment]
Thank you!

**Comment by bagelbits** [comment]
This should be fixed with #26
